src/shared/use-data-value-set/use-data-value-set.js

```diff
@@ -67,6 +72,7 @@ export const useDataValueSet = () => {
         // Only fetch whilst offline, to prevent optimistic updates from being overwritten
         networkMode: 'online',
         meta,
+        onSuccess,
```

Comment:
I initially wanted to populate the has-comment-context directly from this hook (just implement the `onSuccess` handler here), since it is something that is directly related to loading the data values for the dataset. However, I found that when doing it that way the `onSuccess` handler would be called way too often. It turns out this behaviour is intentional in `react-query`, and the reasoning behind it is explained in https://github.com/TanStack/query/issues/1969. By implementing this as a parameter, `onSuccess` is only called by the instance of the hook which passes it, i.e. the one in the data-workspace.

Reply:
I think the update in #123 prevents this, but I'm not totally sure.